java-checks/src/test/java/org/sonar/java/checks/RestrictedIdentifiersUsageCheckTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.java.checks.verifier.JavaCheckVerifier;
+
+import static org.sonar.java.checks.verifier.TestUtils.nonCompilingTestSourcesPath;
+import static org.sonar.java.checks.verifier.TestUtils.testSourcesPath;
+
+class RestrictedIdentifiersUsageCheckTest {
+
+  @Test
+  void test() {
+    JavaCheckVerifier.newVerifier()
+      .onFile(testSourcesPath("checks/RestrictedIdentifiersUsageCheck.java"))
+      .withCheck(new RestrictedIdentifiersUsageCheck())
+      .withJavaVersion(8)
+      .verifyIssues();
+  }
+
+  @Test
+  void test_non_compiling() {
+    JavaCheckVerifier.newVerifier()
+      .onFile(nonCompilingTestSourcesPath("checks/RestrictedIdentifiersUsageCheck.java"))
+      .withCheck(new RestrictedIdentifiersUsageCheck())
+      .withJavaVersion(14)
```

Comment:
Just mentioning: this will not change the actual JDK used, we are always using Java 11 currently. This is usually used when using a `JavaVersionAwareVisitor`. This test makes sense anyway, it makes sure the rule does not change his behavior with a different version.

Reply:
Actually, it sets the version of Java, against which our code will be compiled. And it makes sense, because custom method "yield()" is not allowed to be called outside of the switch expression in Java 13-14, but is absolutely legal in Java 8. And it makes sense to make these versions explicit in tests.

Reply:
Make sense